guillotina/db/storages/pg.py

```diff
@@ -782,21 +782,21 @@ def objects_table_name(self):
 
     @restart_conn_on_exception
     async def open(self):
-        conn = await self.pool.acquire(timeout=self._conn_acquire_timeout)
+        conn = await self.pool.acquire()
```

Comment:
The timeout argument "is not safe". This is implemented in the asyncpg as a `asyncio.wait_for(_acquire(), timeout=timeout)`. The awaitable in `wait_for()` runs in another task. This means that the connections is acquired from the pool in another task and returned to the "parent task". The problem here is that when the parent task is cancelled it won't get the result of the `wait_for()` and leak the connection (because the 'wait_for' task isn't cancelled). 

The issue is reported to asyncpg but isn't solved: https://github.com/MagicStack/asyncpg/pull/468

Reply:
Could we shield the task and in cancelation exception cancel the inner task?

Reply:
Do you mean something like this?
```
try:
    fut = shield(self.pool.acquire(timeout=...))
    await fut
except CancelledError:
    if fut.done():
        # get connection from result and release it
    else:
        fut.cancel()
    raise
```